### ISSUE TITLE
Fine-tune the pre-trained model on my own data?

### ISSUE THREAD

**Issue by CiSong10** [prs-eth/ForAINet#26]
Dear developer,

Could you please provide some guidance on fine-tuning the pre-trained model using my own annotated dataset? Such as, which scripts and config files should I modify, and how? And other things I should be aware of when fine-tuning. Any help or examples would be greatly appreciated. Thanks!

Best,

**Comment by bxiang233** [comment]
> Dear developer,
> 
> Could you please provide some guidance on fine-tuning the pre-trained model using my own annotated dataset? Such as, which scripts and config files should I modify, and how? And other things I should be aware of when fine-tuning. Any help or examples would be greatly appreciated. Thanks!
> 
> Best,

Hi @CiSong10 ,

Basically, you can adjust this parameter to load the pre-trained model and then fine-tune it on your new dataset:
[link to config line](https://github.com/prs-eth/ForAINet/blob/341e46f16884c521233e4457b72acb60e6824736/PointCloudSegmentation/conf/models/panoptic/FORpartseg_3heads.yaml#L177).

For fine-tuning, the easy way is that organizing your new dataset in the same format as FOR-instance. Once the data is in the same format, you can place it into the corresponding folders. [Reference of data folder structure](https://github.com/prs-eth/PanopticSegForLargeScalePointCloud?tab=readme-ov-file#data-folder-structure).

It would be great to hear an update from you if possible.

Best,
Binbin

**Comment by CiSong10** [comment]
Hi Binbin, 

Thank you so much for your instructions! I did perform the fine-tune successfully.

I selected five annotated trees along with their surrounding non-tree background points, and trained, validated, and tested the model on this same dataset. While I knew this setup would cause overfitting, my goal was to perform a sanity check, verifying whether the model has the ability to learn during fine tuning.

I trained the model using the “basic setting” for 80 epochs. Although the training loss fluctuated heavily and did not seem to improve significantly, the predicted results looked ok, and most test metrics improved: test accuracy increased from 83.1 to 92.3, mean IoU from 58.8 to 66.1, and test F1 from 0.80 to 0.97.

**Question**: The paper explores different training settings. Have you evaluated the model’s performance when combining two or more settings (e.g., basic setting + return number + TreeMix)? I am considering adding one or more settings and testing finetuning over a larger area, is there any tips you can offer regarding settings? Much appreciated! 


<img width="601" height="292" alt="Image" src="https://github.com/user-attachments/assets/ff44f463-429c-441d-b8d8-ddf5133f8789" />


<img width="448" height="739" alt="Image" src="https://github.com/user-attachments/assets/777f80ed-d297-4f35-8f03-5284e53d59ac" />

**Comment by bxiang233** [comment]
> Hi Binbin,
> 
> Thank you so much for your instructions! I did perform the fine-tune successfully.
> 
> I selected five annotated trees along with their surrounding non-tree background points, and trained, validated, and tested the model on this same dataset. While I knew this setup would cause overfitting, my goal was to perform a sanity check, verifying whether the model has the ability to learn during fine tuning.
> 
> I trained the model using the “basic setting” for 80 epochs. Although the training loss fluctuated heavily and did not seem to improve significantly, the predicted results looked ok, and most test metrics improved: test accuracy increased from 83.1 to 92.3, mean IoU from 58.8 to 66.1, and test F1 from 0.80 to 0.97.
> 
> **Question**: The paper explores different training settings. Have you evaluated the model’s performance when combining two or more settings (e.g., basic setting + return number + TreeMix)? I am considering adding one or more settings and testing finetuning over a larger area, is there any tips you can offer regarding settings? Much appreciated!
> 
> <img alt="Image" width="601" height="292" src="https://private-user-images.githubusercontent.com/104738913/484752478-ff44f463-429c-441d-b8d8-ddf5133f8789.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY5MDQ4OTMsIm5iZiI6MTc1NjkwNDU5MywicGF0aCI6Ii8xMDQ3Mzg5MTMvNDg0NzUyNDc4LWZmNDRmNDYzLTQyOWMtNDQxZC1iOGQ4LWRkZjUxMzNmODc4OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwM1QxMzAzMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOGNiNGRkOGE4YTUxOTFiZGUzZTA5OWJkYzJmM2FmODUzODUyZjJmYzg0NTg3OWE3ZjhkOGZjZjU3MzAyNTM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.HbDw3LmU1pgg9OgMed6swtdlqzFaX1QxXkcveK1E9NE"> <img alt="Image" width="448" height="739" src="https://private-user-images.githubusercontent.com/104738913/484753246-777f80ed-d297-4f35-8f03-5284e53d59ac.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY5MDQ4OTMsIm5iZiI6MTc1NjkwNDU5MywicGF0aCI6Ii8xMDQ3Mzg5MTMvNDg0NzUzMjQ2LTc3N2Y4MGVkLWQyOTctNGYzNS04ZjAzLTUyODRlNTNkNTlhYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwM1QxMzAzMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mN2IwZmUzZDc2NzdhMjA3OWZlNGE0NmRjNTlmZDNlMzAwZDMzMjNhOWM0M2ZmNTc1ZTAxNzhlN2M5YWE3ZTYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.cUgX8CLAJZa5yDlSx0pOIHVrl_BJLYXiuNAVw-ZlaFA">

Hi @CiSong10 

Regarding your question, yes, I did experiment with combining different settings. However, the combinations didn’t lead to consistent improvements compared to using each setting individually, so I didn’t continue with them.
